xds/csds/csds.go

```diff
@@ -136,8 +137,8 @@ func (s *ClientStatusDiscoveryServer) buildClientStatusRespForReq(req *v3statusp
 // Close cleans up the resources.
 func (s *ClientStatusDiscoveryServer) Close() {
 	s.mu.Lock()
-	if s.xdsClient != nil {
-		s.xdsClient.Close()
+	if s.xdsClientClose != nil {
+		s.xdsClientClose()
 	}
 	s.mu.Unlock()
```

Comment:
Why is this done in a lock?

Is the xdsClient itself not okay with using it while calling close?  Otherwise this `close` doesn't do anything else or indicate that the `xdsClient` no longer exists (e.g. by setting to `nil`), so I don't see why this lock is needed.